File: NET_4.5.2/C1.WPF/VB/C1TreeViewDragDropSample/C1TreeViewDragDropSample2010/Employee.vb
Imports System.Collections.ObjectModel

Namespace C1TreeViewDragDropSample2010
	''' <summary>
	''' Business object used to represent an Employee.
	''' </summary>
	Public Class Employee
		Public Property EmployeeID() As Integer
		Public Property Name() As String
		Public Property Title() As String
		Public Property Gender() As Char
		Public Property ManagerID() As Integer
		Public Property DepartmentID() As Integer
		Public Property Subordinates() As ObservableCollection(Of Employee)

		Public Sub New()
			Subordinates = New ObservableCollection(Of Employee)()
		End Sub

		Public Overrides Overloads Function Equals(ByVal obj As Object) As Boolean
			Dim employee As Employee = TryCast(obj, Employee)
			Dim isEqual As Boolean = False

			If employee IsNot Nothing Then
				isEqual = (employee.EmployeeID = Me.EmployeeID)
			End If
			Return isEqual
		End Function

		Public Function Clone() As Employee
			Return TryCast(Me.MemberwiseClone(), Employee)
		End Function
	End Class
End Namespace

File: NET_4.5.2/C1.WPF/CS/BasicControls/ItemsControls/Employee.cs
using System.Collections.ObjectModel;

namespace BasicControls
{
    /// <summary>
    /// Business object used to represent an Employee.
    /// </summary>
    public class Employee
    {
        public int EmployeeID { get; set; }
        public string Name { get; set; }
        public string Title { get; set; }
        public char Gender { get; set; }
        public int ManagerID { get; set; }
        public int DepartmentID { get; set; }
        public ObservableCollection<Employee> Subordinates { get; set; }

        public Employee()
        {
            Subordinates = new ObservableCollection<Employee>();
        }

        public override bool Equals(object obj)
        {
            Employee employee = obj as Employee;
            bool isEqual = false;

            if (employee != null)
            {
                isEqual = (employee.EmployeeID == this.EmployeeID);
            }
            return isEqual;
        }

        public Employee Clone()
        {
            return this.MemberwiseClone() as Employee;
        }
    }
}
